src/pubsub.js

```diff
@@ -55,29 +52,33 @@ module.exports = (node) => {
     },
 
     publish: (topic, data, callback) => {
-      if (!node.isStarted() && !floodSub.started) {
+      if (!node.isStarted() && !pubsub.started) {
         return nextTick(callback, errCode(new Error(messages.NOT_STARTED_YET), codes.PUBSUB_NOT_STARTED))
       }
 
+      if (!Buffer.isBuffer(data) && typeof data !== 'string') {
+        return nextTick(callback, errCode(new Error('data must be a Buffer or a String'), 'ERR_DATA_IS_NOT_VALID'))
+      }
+
       if (!Buffer.isBuffer(data)) {
-        return nextTick(callback, errCode(new Error('data must be a Buffer'), 'ERR_DATA_IS_NOT_A_BUFFER'))
+        data = Buffer.from(data)
       }
```

Comment:
The buffer check could be simplified by just trying to cast it immediately. So instead of doing both of these checks just do:

```js
try {
  data = Buffer.from(data)
} catch (err) {
  return nextTick(callback, errCode(new Error('data must be convertible to a Buffer'), 'ERR_DATA_IS_NOT_VALID'))
}
```

Reply:
cool! way better